CHANGELOG.md

```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v2.0.0
+
+- Breaking change: must add `priority` column, which requires downtime
```

Comment:
This is not very friendly documentation. Ideally we'd tell the users what will happen if they naively deploy and should tell them how to perform a deployment without throwing lots of errors.

Reply:
Agreed. That said, I'm not sure if the changelog is the place for it. We could add a section to the README?

Also, without pointing someone to the SQL in the `upgrade.ex` file, do you have a suggestion for how best to communicate this information? Or are you thinking we need to explain the process of upgrading their database, generally?

Reply:
It's quite a bit more tricky to perform this database migration as we've written Rihanna to crash unless the user performs the steps in a specific order, and this order isn't the one that the Phoenix docs advise.

For me the ideal solution would be to gracefully handle older versions of the schema (logging 
a warning or error to alert the user to the need to migrate).

If we don't want to do that (as it would be more work for us) we should document the correct process to deploy without causing Rihanna's poller to repeatedly crash while generating lots of errors.

Perhaps we could have an UPDRADING_v2.md that explains the process and reference this in the changelog.

Reply:
I'm basing this as a breaking change of the discussion in the previous PR. I'm definitely not a DBA 😄

I'm not sure I should make any changes to the way we do or do not crash with missing columns at this time.

The only way I can think of to do this change to fit with phoenix/rihanna upgrade path is to have a release to add the columns to the database, and not use them in code. After that, v2 just becomes the addition of the code, which requires the columns/indexes be in place.

Alternatively, I can add the UPGRADING_v2.md file and start adding info there. Off the top of my head:

1. Create a migration, use upgrade
2. Run that migration, without deploying the code, possibly as just a SQL file which we could write
3. Insert the version of the migration into the DB schema_migrations table
4. Deploy code

I'm honestly not sure that `upgrade.ex` really works anyway. E.g., the `down_statements` will drop all columns that have been added, not just the latest column.

Reply:
Also, the upgrade potentially creates the same index 2x on the way up.

Maybe I need to make an upgrade_v2?